projects/electron-ngrx/src/lib/window-communication.service.ts

```diff
@@ -1,59 +1,112 @@
 import {Injectable} from '@angular/core';
 import {fromEvent, Observable} from 'rxjs';
-import {filter, map, withLatestFrom} from 'rxjs/operators';
+import {filter, map, pluck, share, take, withLatestFrom} from 'rxjs/operators';
 import {NavigationEnd, Router} from '@angular/router';
+import {ipcRenderer, remote} from 'electron';
+
+interface WithSender {
+  data: any;
+  senderId: number;
+  massageId: number;
+}
+
+export interface Massage<T> {
+  reply: (...arg: any) => void;
+  data: T;
+}
+
+export const enum chancels {
```

Comment:
Did you mean channels?
I prefer naming enums as a singular and not plural but both are fine.